api/misc.go

```diff
@@ -29,18 +31,65 @@ func (s *Server) eventList(apiContext *api.ApiContext) (*client.GenericCollectio
 	return toEventCollection(eventList), nil
 }
 
-func (s *Server) GenerateSupportBundle(w http.ResponseWriter, req *http.Request) error {
-	bundleFile, fileName, size, err := s.m.GenerateSupportBundle()
+func (s *Server) InitiateSupportBundle(w http.ResponseWriter, req *http.Request) error {
+	var sb *manager.SupportBundle
+	apiContext := api.GetApiContext(req)
+	sb, err := s.m.InitSupportBundle()
 	if err != nil {
+		return errors.Errorf("unable to initiate Support Bundle Download:%v", err)
+	}
+	if sb != nil {
+		apiContext.Write(toSupportBundleResource(s.m.GetCurrentNodeID(), sb))
+	}
+	return nil
+}
+
+func (s *Server) QuerySupportBundle(w http.ResponseWriter, req *http.Request) error {
+	var sbFile SupportBundleQueryInput
```

Comment:
`var query SupportBundleQueryInput`

It can be more than file. So stick with the type name.

Same below.